### DISCUSSION TITLE
Can multiple threads start different containers concurrently?

### DISCUSSION THREAD

**Discussion by cer** [testcontainers/testcontainers-java#6406] [General]
I want to speed up my tests by starting (i.e. `container.start()`) multiple containers concurrently.
Is this supported?
I couldn't find a discussion of thread-safety in the docs.

**Comment by eddumelendez** [comment]
Hi @cer, `Startables.deepStart` can be used. See the code below.

```java
GenericContainer<?> container1 = ...;
GenericContainer<?> container2 = ...;
Startables.deepStart(container1, container2).join();
```

**Reply by cer** [reply]
Cool. It's exactly what I need.

Where's this documented? I just saw [one brief reference](https://www.testcontainers.org/test_framework_integration/junit_5/#extension)

**Reply by cer** [reply]
I assume that it starts a container after its dependencies have become ready, right?

**Reply by cer** [reply]
> I assume that it starts a container after its dependencies have become ready, right?

Apparently, yes!

**Reply by eddumelendez** [reply]
GenericContainer provides `dependsOn()` in order to set the order of execution. I will work on adding documentation.

**Reply by cer** [reply]
This feature is super useful. It speeds up the tests. 

One issue, however, is that there appears to be a risk of overloading the machine:

* 8 core local machine - works fine
* 2 core CI server machine - fails
* 4 core CI server machine - works fine

It might be useful to have a version of `deepStart()` that takes an argument that specifies the max level of concurrency.

**Reply by akinwale-habibullah** [reply]
Thanks for your response @eddumelendez . Is it possible to use similar constructs (Startables) in NodeJS? I have a use case where I also want to start several testcontainers used in different test suites.
